util/fibers/epoll_socket.cc

```diff
@@ -583,55 +582,49 @@ void EpollSocket::Wakey(uint32_t ev_mask, int error, EpollProactor* cntr) {
     ec = make_error_code(errc::connection_aborted);
   }
 
-  if (ev_mask & (EpollProactor::EPOLL_IN | kErrMask)) {
-    if (async_read_pending_) {
-      DCHECK(async_read_req_);
-
-      auto finalize = [this] {
-        delete async_read_req_;
-        async_read_req_ = nullptr;
-        async_read_pending_ = 0;
+  auto body = [&ec, this](bool is_send) {
```

Comment:
since it captures `this`, please make it a member function that returns `error_code`.

Reply:
+1

Reply:
So much better now, I really don't know what I was thinking with this large lambda